### ISSUE TITLE
MALFORMED_REQUEST

### ISSUE THREAD

**Issue by shrimaliakash** [paypal/PayPal-node-SDK#303]
var config = {
    port : 3000,
    api : {
      host : "api.sandbox.paypal.com",
      port : "",
      client_id : "my client_id",   //from paypal account
      client_secret : "my clent_secrete"   //from paypal account
    }
  } 

  paypal.configure(config.api);

  app.post('/api/payments', function(req,res)
  {
    var version = req.params.version;
    var data = {
      intent: 'authenticate',
      redirect_urls:{
        return_url: '/api/payments/success',
        cancel_url: '/api/payments/cancel'
      },
      payer:{
        payment_method: 'paypal'
      },
      transactions:{
        amount:{
          price: req.body.price,
          currency: 'usd'
        },
        name: req.body.item_name,
        description: req.body.description
      }
    };
    console.log(data);
    paypal.payment.create(data, function (error, payment) {
      if (error) {
          console.log(error);
          throw error;
      } else {
        if(data.payer.payment_method == 'paypal') {
          req.paymentId = payment.id;
          var redirectUrl;
          console.log(payment);
          for(var i=0; i < payment.links.length; i++) {
            var link = payment.links[i];
            if (link.method == 'REDIRECT') {
              redirectUrl = link.href;
            }
          }
          res.redirect(redirectUrl);
        }
      }
    });
  });
for that code i have display following error in console:
```
{ intent: 'authenticate',
redirect_urls:
{ return_url: '/api/:version/payments/success',
cancel_url: '/api/:version/payments/cancel' },
payer: { payment_method: 'paypal' },
transactions:
{ amount: { price: '120', currency: 'usd' },
name: 'test',
description: 'nice' } }
{ Error: Response Status : 400
at IncomingMessage.<anonymous> (C:\Users\ketan\Desktop\abc\api\
node_modules\paypal-rest-sdk\lib\client.js:136:23)
at emitNone (events.js:91:20)
at IncomingMessage.emit (events.js:185:7)
at endReadableNT (_stream_readable.js:974:12)
at _combinedTickCallback (internal/process/next_tick.js:80:11)
at process._tickCallback (internal/process/next_tick.js:104:9)
response:
{ name: 'MALFORMED_REQUEST',
message: 'Incoming JSON request does not map to API request',
information_link: 'https://developer.paypal.com/webapps/developer/docs/api/
MALFORMED_REQUEST',
 debug_id: '407caa0e10105',
 httpStatusCode: 400 },
httpStatusCode: 400 }
C:\Users\ketan\Desktop\abc\api\server\boot\routes.js:71
throw error;
^
Error: Response Status : 400
at IncomingMessage.<anonymous> (C:\Users\ketan\Desktop\abc\api\
node_modules\paypal-rest-sdk\lib\client.js:136:23)
at emitNone (events.js:91:20)
at IncomingMessage.emit (events.js:185:7)
at endReadableNT (_stream_readable.js:974:12)
at _combinedTickCallback (internal/process/next_tick.js:80:11)
at process._tickCallback (internal/process/next_tick.js:104:9)
[nodemon] app crashed - waiting for file changes before starting...

>

**Comment by xiaoleih41** [comment]
**Transactions** is an array containing **transaction object**.  Please refer to the SDK [sample](https://github.com/paypal/PayPal-node-SDK/blob/master/samples/payment/create_with_paypal.js) as well as the [API reference/sample](https://developer.paypal.com/docs/api/payments/#payment_create_request) listed on the PayPal developer portal.

**Comment by shrimaliakash** [comment]
this is  not work

**Comment by xiaoleih41** [comment]
What's not working? Please make sure to include **[]** when you refer **transactions** as it's an array.

**Comment by shrimaliakash** [comment]
app.get('/api/webapi/:version/payments/sucess', function(req,res)
  {
    var version = req.params.version;
    console.log(version);
    var paymentId = req.query.paymentId;
    console.log(paymentId);
    var PayerID  = {'payer_id' : req.query.PayerID};
    console.log(PayerID);
    paypal.payment.execute(paymentId, PayerID, function (error, payment) {
      if (error) {
          console.log(error);
          throw error;
      } else {
        if (payment.state === 'approved'){ 
                res.send('payment completed successfully');
                console.log(payment);
            } else {
                res.send('payment not successful');
            }
        // console.log("Get Payment Response");
        // console.log(JSON.stringify(payment));
      }
    });
  });

> 
that code is not work how can you get answer?

**Comment by xiaoleih41** [comment]
I am not sure where you got this code. Are you referring to our sample code? I am not sure what you are trying to accomplish here.  debug_id: 'de4879e85e751' shows that the **payer_Id** is not formatted correctly but I don't have all the details what you have passed on.

**Comment by xiaoleih41** [comment]
Close this issue due to inactivity. Please resubmit when you have additional information.

**Comment by shrimaliakash** [comment]
I have solve this bug player id is declare in the url and that copy and past to the code so work correctly

**Comment by safinn** [comment]
Im getting a 400 response with the name `MALFORMED_REQUEST` when I execute:
```
paypal.payment.execute(paymentId, payerId, (err, payment) => {
    if (err) {
         // Hits here :(
    }
});
```

Any ideas?

**Comment by xiaoleih41** [comment]
Please submit a new issue as this original issue has been closed.
